docs/assignment2.md

```diff
@@ -1 +1,233 @@
+
+
+
+#### **Team**
+
+Class representing a single soccer team. Only exists within a **Game** and is created through the GUI by the user through the selection of 8 **Player** objects.
+
+Attributes 
+
+- *players* (type: Player[]): all the players in a team. Represented by an array as a team has exactly 8 players.
+- *score* (type: int): team's total score
+- *wins* (type: int): team's total number of wins in leagues played by the user; initilized to 0
+- *teamName* (type: String): team name, unique attribute used to identify a team
+
+Operations 
+
+- *int calculateScore()*: calculates the *score* of a team by adding up all the *players* objects individual scores
+-  *int getTeamScore()*: returns *score*
+-  *int getTotalWins()*: returns *wins*
+- *String getTeamName()*: returns *teamName*
+- *void addNewWin()*: increments *wins*
+
+Associations
+
+The composition relation between **Team** and **Game** suggests the fact that all **Team** objects ever created are part of the **Game**.allTeams. 
+Each team is an aggregation of exactly 8 **Player** objects.  
+
+#### **League** 
+
+A league simulates multiple rounds of matches between user-selected teams and determines the winners.
+
+Attributes
+
+- *initialTeams* (type:Queue<Team&gt;): List of the teams playing in the first round, as selected by the user.  Already described in **Game**.createLeague's parameter as: list of teams ordered by their playing order, e.g.  a round with matches (Team1 vs. Team2) and (Team3 vs. Team4) would be represented as {Team1, Team2, Team3, Team4}
+-  *winners* (type: HashMap<Integer, ArrayList<Team&gt;>):  A key represents the round number, e.g. first round is 1, while the values represent the winners at each given round.
+
+Operations
+
+- *void playRound(Queue<Team&gt; teamsToPlay, int round)*: recursive method that generates *winners*. First call is *playRound(initialTeams, 1)*. The method calls itself for following rounds until one team remains (the league winner) 
+- HashMap<Integer, ArrayList<Team>> getWinners(): calls *playRound* method and returns *winners*
+
+Associations
+
+A single **League** object can exist at one time within a **Game**. A league can start with at most 16 teams as per requirement **FR6**. Within the *playRound* method, **Match** objects are created from the playing teams to generate a winner.
+
+#### **Match**
+
+A match simulates a single soccer game between two teams.
+
+Attributes
+
+* *team1* (Type: Team): first team in the match
+* *team2* (Type: Team): second team in the match
+
+Operations
+
+- *Team getWinner()*: returns the winner of a match as described in **FR8** and **FR9**
+
+Assiciations
+
+A match is formed of exactly 2 **Team** objects.
+
+### *Views* package
+
+Each class in this package is an extention of the **GamePane** abstract class, described below. Each uses the *javafx* library and represents a different window in the GUI. All interactions with the *models* package are made via the **Game** class methods.
```

Comment:
**[SEVERE]** Extension is modeled using inheritance arrows (full head arrows), not directed binary association!